### PR TITLE
Added map of s1_area_fraction_top

### PR DESCRIPTION
**Pull request by darrylmasson**
The 3D map of s1 area fraction top is needed to compute the p-values used in the S1AreaFractionTop cut in lax. This version of the map was generated from the 32 keV line in Kr83m, using pax_v6.4.2.

### CODE REVIEW

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import numpy as np
+
+import json
+
+from pax import plugin, exceptions
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [F401] 'exceptions' imported but unused

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from pax.datastructure import Interaction
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [F401] 'Interaction' imported but unused

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+from scipy.interpolate import RectBivariateSpline
+from scipy.stats import binom_test
+
+class S1AreaFractionTopPvalue(plugin.TransformPlugin):
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    """Computes p-value for S1 area fraction top
+
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [W293] blank line contains whitespace

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+from scipy.interpolate import RectBivariateSpline
+from scipy.stats import binom_test
+
+class S1AreaFractionTopPvalue(plugin.TransformPlugin):
+    """Computes p-value for S1 area fraction top
+    
+    """
+
+    def startup(self):
+        aftmap_filename = utils.data_file_name('s1_aft_rz_02Mar2017.json')
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [F821] undefined name 'utils'

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        with open(aftmap_filename) as data_file:
+            data = json.load(data_file)
+        r_pts = np.array(data['r_pts'])
+        z_pts = np.array(data['z_pts'])
+        aft_vals = []
+        for i in range(len(r_pts)): # unpacks 1d array to 2d
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [E261] at least two spaces before inline comment

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            aft_vals.append(data['map'][i*len(z_pts):(i+1)*len(z_pts)])
+        aft_vals = np.array(aft_vals)[0,0]
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [E231] missing whitespace after ','

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        self.aft_map = RectBivariateSpline(r_pts,z_pts,aft_vals)
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [E231] missing whitespace after ','

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+    def transform_event(self, event):
+
+        for ia in event.interactions:
+            s1 = event.peaks[ia.s1]
+            s2 = event.peaks[ia.s2]
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [F841] local variable 's2' is assigned to but never used

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [W293] blank line contains whitespace

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    def transform_event(self, event):
+
+        for ia in event.interactions:
+            s1 = event.peaks[ia.s1]
+            s2 = event.peaks[ia.s2]
+            
+            s1_top = np.round(s1.area*s1.area_fraction_top)
+            s1_tot = np.round(s1.area)
+            aft = self.aft_map(np.sqrt(ia.x**2 + ia.y**2),ia.z)
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [E231] missing whitespace after ','

File: pax/datastructure.py [hunk 12]
```diff
@@ -584,6 +584,9 @@ def log_cs2_cs1(self):
     #: Goodness of fit of S1 pattern to interaction's (x, y, drift_time),
     #: computed using number of hits/channel instead of area/channel
     s1_pattern_fit_hits = float('nan')
+
```
Comment (tunnell) [inline pax/datastructure.py]: __[flake8]__
* [W293] blank line contains whitespace

File: pax/plugins/interaction_processing/S1AFTPvalue.py [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+import json
```
Comment (tunnell) [inline pax/plugins/interaction_processing/S1AFTPvalue.py]: __[flake8]__
* [F401] 'json' imported but unused

File: pax/plugins/interaction_processing/S1AreaFractionTopProbability.py [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+        for ia in event.interactions:
+            s1 = event.peaks[ia.s1]
+
+            if s1.area < self.low_pe_threshold:
+                s1_frac = s1.area/self.low_pe_threshold
+                hits_top = s1.n_hits*s1.hit_fraction_top
+                s1_top = s1.area*s1.area_fraction_top
+                size_top = np.round(hits_top*(1-s1_frac) + s1_top*s1_frac)
+                size_tot = np.round(s1.n_hits*(1-s1_frac) + s1.area*s1_frac)
```
Comment (pdeperio) [inline pax/plugins/interaction_processing/S1AreaFractionTopProbability.py]: @darrylmasson Why are you adding hits and area (different units) here? Am I missing a description of this in [your note](https://xecluster.lngs.infn.it/dokuwiki/doku.php?id=xenon:xenon1t:darryl:xe1t_s1_aft_map)?

**Pull request merged**